src/status_im/ui2/screens/chat/components/contact_item/view.cljs

```diff
@@ -9,26 +9,43 @@
             [quo2.components.markdown.text :as text]
             [status-im.ui2.screens.chat.components.message-home-item.style :as style]
             [utils.re-frame :as rf]
-            [status-im.ui2.screens.chat.actions :as actions]))
+            [status-im.ui2.screens.chat.actions :as actions]
+            [quo2.components.selectors.selectors :as selectors]))
 
 (defn open-chat [chat-id]
-  (rf/dispatch [:dismiss-keyboard])
-  (if platform/android?
-    (rf/dispatch [:chat.ui/navigate-to-chat-nav2 chat-id])
-    (rf/dispatch [:chat.ui/navigate-to-chat chat-id]))
-  (rf/dispatch [:search/home-filter-changed nil])
-  (rf/dispatch [:accept-all-activity-center-notifications-from-chat chat-id]))
+  (let [view-id (rf/sub [:view-id])]
+    (when (= view-id :shell-stack)
+      (rf/dispatch [:dismiss-keyboard])
+      (if platform/android?
+        (rf/dispatch [:chat.ui/navigate-to-chat-nav2 chat-id])
+        (rf/dispatch [:chat.ui/navigate-to-chat chat-id]))
+      (rf/dispatch [:search/home-filter-changed nil])
+      (rf/dispatch [:accept-all-activity-center-notifications-from-chat chat-id]))))
 
-(defn contact-item [item]
+(defn action-icon [{:keys [public-key] :as item} {:keys [icon group added] :as extra-data}]
+  (let [{:keys [contacts]} group
+        member? (contains? contacts public-key)]
+    [rn/touchable-opacity {:on-press #(rf/dispatch [:bottom-sheet/show-sheet
+                                                    {:content (fn [] [actions/actions item extra-data])}])
+                           :style {:position :absolute
+                                   :right    20}}
+     (if (= icon :options)
+       [icons/icon :i/options {:size 20 :color (colors/theme-colors colors/neutral-50 colors/neutral-40)}]
+       [selectors/checkbox {:default-checked? member?
+                            :on-change        (fn [selected] (if selected
+                                                               (swap! added conj public-key)
+                                                               (reset! added (filter #(-> % (not= public-key)) @added))))}])]))
```

Comment:
Unneeded thread first macro

Reply:
You can also use `remove` to simplify the filtering step.

```clojure
(reset! added (remove #(= % public-key) @added))
```

And you might like to use `partial` because we can't nest the `#()` reader macro, but some people find it difficult to read, so use sparingly:

```clojure
(swap! added #(remove (partial = public-key) %))
```

Reply:
@ilmotta so both these statements does the same thing?

Reply:
Hey @OmarBasem 
I mean `#(-> % (not= public-key))` is more readable as `#(not= % public-key)`

Reply:
> @ilmotta so both these statements does the same thing?

Exactly @OmarBasem, it's up to the person to decide if they like the second way. I prefer to usually avoid `partial` and use the literal function `#()` form, but `partial` is a good tool to have at hand ;)